pkg/kubelet/kubelet.go

```diff
@@ -2360,18 +2361,20 @@ func (kl *Kubelet) updateRuntimeUp() {
 }
 
 func (kl *Kubelet) updateOneRuntimeUp(s *kubecontainer.RuntimeStatus) error {
-	msg := "either runtime or network is not ready. flag caller to return"
+	msg := "runtime is not ready. flag caller to return"
 	kl.updateRuntimeMux.Lock()
 	defer kl.updateRuntimeMux.Unlock()
 
 	// TODO(random-liu): Consider to send node event when optional
 	// condition is unmet.
 	klog.V(4).Infof("Container runtime status: %v", s)
+
 	networkReady := s.GetRuntimeCondition(kubecontainer.NetworkReady)
 	if networkReady == nil || !networkReady.Status {
+		// do not return error when network is not ready yet, as it would prevent cni pod from starting,
+		// which forms a dead lock in bootstrapping. Logging is just fine.
 		klog.Errorf("Container runtime network not ready: %v", networkReady)
 		kl.runtimeState.setNetworkState(fmt.Errorf("runtime network not ready: %v", networkReady))
-		return fmt.Errorf(msg)
```

Comment:
if this does't return and the compute condition is ready, it will lead the line 2358 and 2359 t be run for each loop in the caller function. could be a potential issue if so.

Reply:
this fix is specifically about the dead lock as reported in #595. The dead lock was formed because (s *runtimeState) runtimeErrors() returned error, as there was no lastBaseRuntimeSync timestamp refreshed.

In order to break the dead lock, we need to ensure runtimeState.lastBaseRuntimeSync is touched. The fix is to exactly ensure line 2358 and 2359 be execute, to touch the time stamp, so to break the lock.

There may exist more optimal frequency refreshing the timestamp; however, it seems not a real concern right now, unless it really impacts system performance.

Reply:
sounds good. I just checked the community version of this function they does the same thing for only return the function when the compute is not ready. we should be good here.

Reply:
Exactly, this regression bug was introduced in commit 1e698f6.
```golang
        if networkReady == nil || !networkReady.Status {
                klog.Errorf("Container runtime network not ready: %v", networkReady)
                kl.runtimeState.setNetworkState(fmt.Errorf("runtime network not ready: %v", networkReady))
+               return fmt.Errorf(msg)
        } else {
```